### PR TITLE
Added endDate helper for ATG

### PR DESCRIPTION
**Pull request by sam-cook**
#### What are the links to relevant tickets?
https://hxshortbreaks.atlassian.net/browse/WEB-6569
#### What does this PR do?
Adds a helper to Dust JS that calculates a customer's check out date based on their check in date and nights stayed
#### What functional/unit tests does this PR have?
I've added unit tests for this piece of work under: tests/calculateEndDateTest.js
#### How should a developer review this?
Pull down and test
#### Any background context you want to provide?
Related to: https://github.com/holidayextras/theatreBlueprint/pull/115
and: https://github.com/holidayextras/the-wall/pull/77
#### Screenshots (if appropriate)
---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [ ] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @t1gr0u
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2 @taijuten
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

File: helpers/calcEndDate.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* jslint node: true */
```
Comment (t1gr0u) [inline helpers/calcEndDate.js]: can the name of the file and the dust helper be `calculateEndDate`

File: test/calcEndDateTest.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/* jslint node: true */
+'use strict';
+
+var chai = require( 'chai' );
+var chaiAsPromised = require( 'chai-as-promised' );
+chai.use( chaiAsPromised );
```
Comment (t1gr0u) [inline test/calcEndDateTest.js]: `chai.use( require( 'chai-as-promised' ) );`

File: helpers/calcEndDate.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  /*
+  * @description Calculate a customer's check-out date based on their check-in date and the amount of nights they have booked
+  * @param {string} startDate start date
+  * @param {string} nights amount of nights
+  * @param {string} [type=days] the measurement of time to add to
+  * @example {@_calcEndDate startDate="07/01/2015" nights="3" type="days" /} output "10/01/2015"
+  */
+
+  dust.helpers._calcEndDate = function( chunk, context, bodies, params ) {
+    if (!params) {
```
Comment (lukefrake) [inline helpers/calcEndDate.js]: Is it worth checking if nights is a positive number?

**Comment by notABluesSinger** [comment]
@sam-cook taking a look at this PR and it occurred to me that this could be a more generic helper, rather than being bound to this particular use case.
What you have here is a date manipulation helper, and by simply adding another parameter for the required operation (`add`, `subtract` etc.) you could make this much more generic and potentially useful in other places.
In order to be completely generic you would need to rename the file and function as well as increases the scope of the tests, but you would end up with a more useful helper.
Thoughts?

**Comment by sam-cook** [comment]
@notABluesSinger happy to do that at a later date after this urgent change has gone live and after we finish current ATG sprint :) Will update this with link to a jira for making this helper more generic

File: helpers/calculateEndDate.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  /*
+  * @description Calculate a customer's check-out date based on their check-in date and the amount of nights they have booked
+  * @param {string} startDate start date
+  * @param {string} nights amount of nights
+  * @param {string} [type=days] the measurement of time to add to
+  * @example {@_calcEndDate startDate="07/01/2015" nights="3" type="days" /} output "10/01/2015"
+  */
+
+  dust.helpers._calcEndDate = function( chunk, context, bodies, params ) {
```
Comment (t1gr0u) [inline helpers/calculateEndDate.js]: Need to change the name to `dust.helpers._calculateEndDate`

**Pull request merged**